pygfx/utils/show.py

```diff
@@ -16,9 +17,243 @@
+        camera=None,
+        before_render=None,
+        after_render=None,
+        draw_function=None,
+    ) -> None:
+        self.canvas = canvas
+        self.renderer = renderer
+        self.controller = controller
+        self.camera = camera
+        self.draw_function = draw_function or self.default_draw
+        self.scene = None
+
+        if before_render is not None:
+            self.before_render = before_render
+
+        if after_render is not None:
+            self.after_render = after_render
+
+    def default_draw(self):
+        if self.before_render is not None:
+            self.before_render()
+
+        self.controller.update_camera(self.camera)
+        self.renderer.render(self.scene, self.camera)
+
+        if self.after_render is not None:
+            self.after_render()
+
+        self.renderer.request_draw()
+
+    def show(
+        self,
+        object: WorldObject,
+        up=None,
+    ):
+        """Display a WorldObject
+
+        This function provides you with the basic scaffolding to visualize a given
+        WorldObject in a new window. While it does add scaffolding, it aims to be
+        fully customizable so that you can replace each piece as needed.
+
+        Parameters
+        ----------
+        object : gfx.WorldObject
+            The object to show. If it is not a :class:`gfx.Scene <pygfx.Scene>`
+            then Display will wrap it into a new scene containing lights and a
+            background.
+        up : gfx.Vector3
+            If set, and ``object`` does not contain a controller, set the camera
+            controller's up vector to this value.
+
+        Notes
+        -----
+        If you want to display multiple objects prefer :class:`gfx.Group
+        <pygfx.Group>` over :class:`gfx.Scene <pygfx.Scene>` to avoid the pitfall of
+        forgetting to add lights to the scene.
+
+        """
+
+        if isinstance(object, Scene):
+            scene = object
+        else:
+            scene = Scene()
+            scene.add(object)
+
+            dark_gray = np.array((169, 167, 168, 255)) / 255
+            light_gray = np.array((100, 100, 100, 255)) / 255
+
+            background = Background(None, BackgroundMaterial(light_gray, dark_gray))
+            scene.add(background)
+            scene.add(AmbientLight(), DirectionalLight())
+        self.scene = scene
+
+        if len(self.find_children(Light)) == 0:
+            warnings.warn(
+                "Your scene does not contain any lights. Some objects may not be visible"
+            )
+
+        existing_cameras = self.find_children(Camera)
+        if self.camera:
+            pass
+        elif len(existing_cameras) > 0:
+            self.camera = existing_cameras[0]
+        elif not self.camera:
+            self.camera = PerspectiveCamera(70, 16 / 9)
+            self.camera.add(DirectionalLight())
+            # do we really want to modify the scene and add
+            # a camera?
```

Comment:
If you don't do this, the directional light is not in the scene either :)

Reply:
Yes. my question is more from the angle of "do we want to modify the input scene?". It's similar to our discussion about adding lights by default: How worried are we about the side effects of permanently modifying the user's scene?

Reply:
Yes, well, I think we had that discussion earlier, I think all proposals are still on the table. Let's just go with this approach for now and adjust it later if we think of something better.